### PR TITLE
Add more checks 

### PR DESCRIPTION
**Pull request by cosmatudor**

### CODE REVIEW

File: integrationTests/relayers/slowTests/framework/ethereumHandler.go [hunk 1]
```diff
@@ -277,14 +277,22 @@ func (handler *EthereumHandler) IssueAndWhitelistToken(ctx context.Context, para
...
+		if params.IsMintBurnOnEth {
+			var tx2 *types.Transaction
+			var err2 error
+			if params.IsNativeOnEth {
+				burnAmount := initialSupply
+				tx2, err2 = handler.SafeContract.InitSupplyMintBurn(auth, erc20Address, zeroValueBigInt, burnAmount)
+			} else {
+				mintAmount := initialSupply
+				tx2, err2 = handler.SafeContract.InitSupplyMintBurn(auth, erc20Address, mintAmount, zeroValueBigInt)
+			}
```
Comment (sstanculeanu) [inline integrationTests/relayers/slowTests/framework/ethereumHandler.go:292]: perhaps extract `txInitSupplyMintBurn, errInitSupplyMintBurn := handler.SafeContract.InitSupplyMintBurn(auth, erc20Address, mintAmount, burnAmount)` to be called only once after this if and only set minAmount and burnAmount here

File: integrationTests/relayers/slowTests/framework/multiversxHandler.go [hunk 2]
```diff
@@ -800,21 +803,41 @@ func (handler *MultiversxHandler) setInitialSupply(ctx context.Context, params I
...
+					initSupplyMintBurnEsdtSafe,
+					[]string{
+						hex.EncodeToString([]byte(tkData.MvxChainSpecificToken)),
+						hex.EncodeToString(mintAmount.Bytes()),
+						hex.EncodeToString(zeroValueBigInt.Bytes()),
+					},
+				)
+				log.Info("initial supply tx executed", "hash", hash, "status", txResult.Status,
+					"initial mint", params.InitialSupplyValue, "initial burned", "0")
+			}
```
Comment (sstanculeanu) [inline integrationTests/relayers/slowTests/framework/multiversxHandler.go:840]: same here, you can call ScCall only once, just set the proper variables inside ifs

File: integrationTests/relayers/slowTests/framework/multiversxHandler.go [hunk 3]
```diff
@@ -1043,6 +1066,33 @@ func (handler *MultiversxHandler) TransferToken(ctx context.Context, source Keys
 		"hash", hash, "status", txResult.Status)
 }
 
+func (handler *MultiversxHandler) GetTotalBalancesForToken(ctx context.Context, token string) *big.Int {
```
Comment (sstanculeanu) [inline integrationTests/relayers/slowTests/framework/multiversxHandler.go:1069]: missing comments on new methods

File: integrationTests/relayers/slowTests/framework/multiversxHandler.go [hunk 4]
```diff
@@ -1043,6 +1066,33 @@ func (handler *MultiversxHandler) TransferToken(ctx context.Context, source Keys
...
+	queryParams := []string{
+		hex.EncodeToString([]byte(token)),
+	}
+	responseData := handler.ChainSimulator.ExecuteVMQuery(ctx, handler.SafeAddress, getTotalBalances, queryParams)
+	value := big.NewInt(0).SetBytes(responseData[0])
```
Comment (sstanculeanu) [inline integrationTests/relayers/slowTests/framework/multiversxHandler.go:1074]: is there any chance that responseData is nil here, resulting in a panic while trying to access index 0?

File: integrationTests/relayers/slowTests/framework/multiversxHandler.go [hunk 5]
```diff
@@ -803,41 +803,31 @@ func (handler *MultiversxHandler) setInitialSupply(ctx context.Context, params I
...
-						hex.EncodeToString(mintAmount.Bytes()),
-						hex.EncodeToString(zeroValueBigInt.Bytes()),
-					},
-				)
-				log.Info("initial supply tx executed", "hash", hash, "status", txResult.Status,
-					"initial mint", params.InitialSupplyValue, "initial burned", "0")
+				mintAmount = initialSupply
 			}
+
+			hash, txResult := handler.ChainSimulator.ScCall(
```
Comment (iulianpascalau) [inline integrationTests/relayers/slowTests/framework/multiversxHandler.go:815]: :+1:

**Review by sstanculeanu — DISMISSED** [review]

File: integrationTests/relayers/slowTests/framework/testSetup.go [hunk 6]
```diff
@@ -461,45 +469,54 @@ func (setup *TestSetup) IsTransferDoneFromMultiversX(sender KeysHolder, receiver
...
 	if params.IsMintBurnOnMvX {
-		if !setup.checkMvxBurnedTokenBalance(params) {
-			return false
-		}
-	} else if !setup.checkMvxLockedBalanceForToken(params, false) {
-		return false
+		return setup.checkMvxBurnedTokenBalance(params)
 	}
 
+	return setup.checkMvxLockedBalanceForToken(params, secondBridge)
```
Comment (iulianpascalau) [inline integrationTests/relayers/slowTests/framework/testSetup.go:508]: mismatch between the function name and this provided parameter. Also on the next function

**Review by sstanculeanu — APPROVED** [review]

File: integrationTests/relayers/slowTests/framework/testSetup.go [hunk 7]
```diff
@@ -357,19 +469,124 @@ func (setup *TestSetup) IsTransferDoneFromMultiversX(sender KeysHolder, receiver
...
+func (setup *TestSetup) getMvxTotalRefundAmountForToken(params TestTokenParams) *big.Int {
+	totalRefund := big.NewInt(0)
+	for _, operation := range params.TestOperations {
+		if len(operation.MvxSCCallData) > 0 || operation.MvxForceSCCall {
+			if operation.MvxFaultySCCall {
+				// the balance should be bridged back to the receiver on Ethereum - fee
+				totalRefund.Add(totalRefund, operation.ValueToTransferToMvx)
+				totalRefund.Sub(totalRefund, feeInt)
+			}
+		}
```
Comment (sstanculeanu) [inline integrationTests/relayers/slowTests/framework/testSetup.go:587]:
```suggestion
		if !operation.MvxForceSCCall{
			continue
		}
		if len(operation.MvxSCCallData) == 0l {
			continue
		}
		if !operation.MvxFaultySCCall {
			continue
		}

		// the balance should be bridged back to the receiver on Ethereum - fee
		totalRefund.Add(totalRefund, operation.ValueToTransferToMvx)
		totalRefund.Sub(totalRefund, feeInt)
```

Comment (cosmatudor) [reply integrationTests/relayers/slowTests/framework/testSetup.go]: I'm not sure these 2 versions behave the same. The condition from this if:
`if len(operation.MvxSCCallData) > 0 || operation.MvxForceSCCall {`
is passed if one of those two assertions is true.

On the other hand, using ifs in the suggested order behaves as if both conditions (operation.MvxForceSCCall being true and len(operation.MvxSCCallData) > 0) must be satisfied simultaneously

Working to refactor this logic

**Review by sstanculeanu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**